### ISSUE TITLE
[onert/train] all of backend contexts are candidates when `TrainableExecutor` is created

### ISSUE THREAD

**Issue by YongseopKim** [Samsung/ONE#12641]
When `TrainableExecutor` is created, all of backend contexts are candidates.

In `ExecutorFactory::createTrainableExecutor`, 

https://github.com/Samsung/ONE/blob/890b72859b19af9c29488f88fc2e927a75f35c77/runtime/onert/core/src/compiler/ExecutorFactory.cc#L673-L681

The only candidate should be `train` backend context.

/cc @Samsung/one_onert

**Comment by jyoungyun** [comment]
I agree with you on this point. The `TrainableExecutor` must have a `train` backend context. However, I don't know it is enough to train the model using only the train backend context.

It seems necessary to restrict `TrainableExecutor` to include `Train` backend.

**Comment by hseok-oh** [comment]
Maybe for partial training?

**Comment by YongseopKim** [comment]
(Anyway, my problem was resolved.)
